### PR TITLE
#166729093 Enable user to login

### PR DESCRIPTION
**Pull request by rukundoeric**
### What does this PR do?
* Enable user to login 
### Description of Task to be completed?
  Have these end point working
 * `POST /api/auth/login`
### How should this be manually tested?
 * pull this branch
 * Add some records in your Users table in database.
 * And the test that api above with postman
### Any background context you want to provide?

N/A

### What are the relevant pivotal tracker stories?

[#166729093](https://www.pivotaltracker.com/story/show/166729093)

####  Invalid inputs
Screen Shot 2019-06-17 at 14 48 25

 #### User not exist
Screen Shot 2019-06-17 at 14 48 04

#### On successful operation 
Screen Shot 2019-06-17 at 14 47 21

#### Use case Documentation
[Use case ](https://docs.google.com/document/d/11TbSjNqM6Amt7ZC4aKcDw2EUZ1VuYQlg-ERBUjuMpq8/edit?usp=sharing)

### Questions:
N/A

### CODE REVIEW

File: src/api/controllers/auth.js [hunk 1]
```diff
@@ -132,6 +137,61 @@ class AuthController {
...
+  /**
+   * signup controller
+   * @param {Object} req  - Response
+   * @param {Object} res  - Response
+   * @param {Function} next -Next
+   * @returns {Object} The response object
+   */
+  static async login(req, res) {
+    joi
+      .validate(req.body, Validation.loginShema)
```
Comment (eliemugenzi) [inline src/api/controllers/auth.js]: Could you do this in middlewares?

File: src/api/controllers/auth.js [hunk 2]
```diff
@@ -132,6 +137,61 @@ class AuthController {
...
+      .then(() => {
+        User.findAll({
```
Comment (eliemugenzi) [inline src/api/controllers/auth.js]: It could be better if you use async/await

**Review by EmyRukundo — APPROVED** [review]

File: src/api/controllers/auth.js (hunk 1 above)
Comment (ghost) [reply src/api/controllers/auth.js]: There is a middleware for body validation already. You can use it. Put your schema in `helpers/validationSchemas.js` and use it from `middleware/validateBody.js`

File: src/api/controllers/auth.js [hunk 3]
```diff
@@ -130,6 +130,61 @@ class AuthController {
...
+                message: 'User with that email does not exist.'
+              }
+            });
+          }
+        });
+      })
+      .catch(error => res.status(400).send({
+        status: 400,
+        error: {
+          message: error.message.replace(/[&/\\#,+()$~%.'":*?<>{}]/g, '')
```
Comment (ghost) [inline src/api/controllers/auth.js]: You won't need to do this if you use the `validateBody.js` middleware

File: src/api/routes/authRouter.js [hunk 4]
```diff
@@ -45,6 +46,7 @@ authRouter.get('/login/twitter', passport.authenticate('twitter', { scope: ['pro
 authRouter.get('/login/twitter/redirect', passport.authenticate('twitter', { session: false }), twitter, socialLogin.twitterLogin);
 
 authRouter.post('/signup', validateBody('signup'), validateGender, usernameExists, emailExists, register);
+authRouter.post('/login', login);
```
Comment (ghost) [inline src/api/routes/authRouter.js]: You can make use of the middleware here like this: `validateBody('signin')`. Where `'signin'` is a schema that will be added to `helpers/validationSchemas.js`. I explained this on the slack channel but I am willing to help if you need it.

File: src/middleware/authValidation.js [hunk 5]
```diff
@@ -6,5 +6,9 @@ export default {
   }),
   applyPasswordShema: joi.object().keys({
     newpassword: joi.string().regex(/^[a-zA-Z]/).min(8).required(),
+  }),
+  loginShema: joi.object().keys({
```
Comment (ghost) [inline src/middleware/authValidation.js]: These schemas should be in `helpers/validationSchemas.js`.

**Review by ghost — CHANGES_REQUESTED** [review]

File: .env.example [hunk 6]
```diff
@@ -15,4 +15,4 @@ TWITTER_APP_SECRET=
 
 GOOGLE_ID=
 GOOGLE_SECRET=
-SECRET_KEY =
```
Comment (mifeille) [inline .env.example:18]: this SECRET_KEY should not be removed cause it is used by the JWT
and SECRET is for social login

**Review by UhiriweAudace — APPROVED** [review]

**Review by eliemugenzi — APPROVED** [review]

**Pull request merged**